applications/GeoMechanicsApplication/custom_conditions/T_condition.cpp

```diff
@@ -0,0 +1,141 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#include "custom_conditions/T_condition.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition() : Condition()
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition(IndexType NewId, GeometryType::Pointer pGeometry)
+    : Condition(NewId, pGeometry)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition(IndexType NewId,
+                                        GeometryType::Pointer pGeometry,
+                                        PropertiesType::Pointer pProperties)
+    : Condition(NewId, pGeometry, pProperties)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::~TCondition() = default;
+
+template <unsigned int TDim, unsigned int TNumNodes>
+Condition::Pointer TCondition<TDim, TNumNodes>::Create(IndexType NewId,
+                                                       NodesArrayType const& rThisNodes,
+                                                       PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(
+        new TCondition(NewId, GetGeometry().Create(rThisNodes), pProperties));
```

Comment:
I think here we should be able to use Kratos::make_intrusive. Condition::Pointer is of the type Kratos::intrusive_ptr, which is probably a custom implementation of the shared_ptr (at some point I think this should become a shared_ptr, but since that's in core, that will take some time to do). However, it also has this Kratos::make_intrusive<T>() function, which allows us to avoid using 'new'
```suggestion
    return Kratos::make_intrusive<TCondition>(NewId, GetGeometry().Create(rThisNodes), pProperties);
```

Reply:
Thank you for the info. It is implemented.